jablib/src/main/java/org/jabref/logic/citation/repository/MVStoreBibEntryRelationRepository.java

```diff
@@ -0,0 +1,268 @@
+        this.store = this.storeConfiguration.open();
+    }
+
+    @Override
+    public List<BibEntry> getRelations(BibEntry entry) {
+        return entry
+            .getDOI()
+            .map(doi -> {
+                MVMap<String, LinkedHashSet<BibEntry>> relationsMap = store.openMap(mapName, mapConfiguration);
+                return relationsMap.getOrDefault(doi.asString(), new LinkedHashSet<>()).stream().toList();
+            })
+            .orElse(List.of());
+    }
+
+    /**
+     * Allows insertion of empty list in order to keep track of insertion date for an entry.
+     */
+    @Override
+    synchronized public void addRelations(@NonNull BibEntry entry, @NonNull List<BibEntry> relations) {
+        entry.getDOI().ifPresent(doi -> {
+            // Save the relations
+            MVMap<String, LinkedHashSet<BibEntry>> relationsMap = this.store.openMap(mapName, mapConfiguration);
+            var relationsAlreadyStored = relationsMap.getOrDefault(doi.asString(), new LinkedHashSet<>());
+            relationsAlreadyStored.addAll(relations);
+            relationsMap.put(doi.asString(), relationsAlreadyStored);
+
+            // Save insertion timestamp
+            var insertionTime = LocalDateTime.now(TIME_STAMP_ZONE_ID);
+            MVMap<String, LocalDateTime> insertionTimeStampMap = store.openMap(insertionTimeStampMapName);
+            insertionTimeStampMap.put(doi.asString(), insertionTime);
+
+            // Commit
+            store.commit();
+        });
+    }
+
+    @Override
+    synchronized public boolean containsKey(BibEntry entry) {
+        return entry
+            .getDOI()
+            .map(doi -> {
+                MVMap<String, LinkedHashSet<BibEntry>> relationsMap = store.openMap(mapName, mapConfiguration);
+                return relationsMap.containsKey(doi.asString());
+            })
+            .orElse(false);
+    }
+
+    @Override
+    synchronized public boolean isUpdatable(BibEntry entry) {
+        var clock = Clock.system(TIME_STAMP_ZONE_ID);
+        return this.isUpdatable(entry, clock);
+    }
+
+    @VisibleForTesting
+    boolean isUpdatable(final BibEntry entry, final Clock clock) {
+        final var executionTime = LocalDateTime.now(clock);
+        return entry
+            .getDOI()
+            .map(doi -> {
+                MVMap<String, LocalDateTime> insertionTimeStampMap = store.openMap(insertionTimeStampMapName);
+                return insertionTimeStampMap.getOrDefault(doi.asString(), executionTime);
+            })
+            .map(lastExecutionTime -> lastExecutionTime.equals(executionTime)
+                || lastExecutionTime.isBefore(executionTime.minusDays(this.storeTTLInDays))
+            )
+            .orElse(true);
+    }
+
+    @Override
+    public void close() {
+        this.store.close();
+    }
+
+    static class BibEntrySerializer extends BasicDataType<BibEntry> {
```

Comment:
I think, this class is wrong - one should use code similar as `org.jabref.gui.ClipBoardManager#serializeEntries`

Reason: `The result is close to BibTeX, but not a valid BibTeX representation in all cases` - see documentation of the used `org.jabref.model.entry.CanonicalBibEntry#getCanonicalRepresentation`